### PR TITLE
Add initial Django app and XForm create view & Docker

### PR DESCRIPTION
**Pull request by nkhumphreys**
This commit contains the initial django setup for the odk_importer
(replacement for formhub) and a working view (with some TODO's) for
uploading an XML file which contains an XForm.  No validation is
currently done on the xml in the file

### CODE REVIEW

File: gather2_odk_importer/requirements_importer.txt [hunk 1]
```diff
@@ -0,0 +1,4 @@
+Django==1.8.6
+djangorestframework==3.3.1
```
Comment (nkhumphreys) [inline gather2_odk_importer/requirements_importer.txt]: This needs removing

File: gather2_odk_importer/requirements_importer.txt [hunk 2]
```diff
@@ -0,0 +1,4 @@
+Django==1.8.6
```
Comment (nkhumphreys) [inline gather2_odk_importer/requirements_importer.txt]: Will upgrade this to 1.9

**Comment by nkhumphreys** [comment]
flake8 is failing but this is only because its a WIP that contains some files generated by Django that I just have not go to yet

File: gather2_odk_importer/importer/settings.py [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+
+USE_L10N = True
+
+USE_TZ = True
+
+
+# Static files (CSS, JavaScript, Images)
+# https://docs.djangoproject.com/en/1.8/howto/static-files/
+
+STATIC_URL = '/static/'
```
Comment (readevalprint) [inline gather2_odk_importer/importer/settings.py]: Could you add this to the end and not use `local.py`? It's subjective, I know, but I want to standardize on One True Way™ and it should work by default out of the box w/o special settings for development. And anything you woudl want to put in local settings should also have pull from the environment. And could you test using the postgres container specified in `docker-compose.yml`

```
DATABASES = {
    'default': {
        'ENGINE': 'django.db.backends.postgresql_psycopg2',
        'NAME': os.environ.get('DATABASE_NAME', 'odkimporter'),
        'USER': os.environ.get('DATABASE_USER', 'odkimporter'),
        'HOST': os.environ.get('DATABASE_HOST', 'localhost'),
        'PORT': os.environ.get('DATABASE_PORT', '5432'),
    }
}

try:
    from local_settings import *  # noqa
except ImportError as e:
    logger.error(e)
```

**Comment by readevalprint** [comment]
Also could you rename `gather2_odk_importer` to something using dashes instead of underscores. It was like that originally to reduce the urge to import this folder. This should be a stand alone server.

**Comment by nkhumphreys** [comment]
@readevalprint all your comments should now be addressed and PEP8 checks should pass

File: gather2-odk-importer/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,29 @@
+FROM ubuntu:14.04
+
+ENV HOME /root
+ENV DEBIAN_FRONTEND noninteractive
+
+RUN apt-get install -y wget
+RUN wget --quiet -O - https://www.postgresql.org/media/keys/ACCC4CF8.asc | apt-key add -
+RUN echo "deb http://apt.postgresql.org/pub/repos/apt/ $(lsb_release -cs)-pgdg main 9.4" > /etc/apt/sources.list.d/pgdg.list
+
+ADD requirements_importer.txt /tmp/requirements.txt
```
Comment (readevalprint) [inline gather2-odk-importer/Dockerfile]: Optimization for the future, if you add `requirements.txt` right before it's pip installed, when it's invalidated only the pip install will be done again. As is stands, the apt-get will be re run as well even though the ubuntu packages may not have changed.

Comment (nkhumphreys) [inline gather2-odk-importer/Dockerfile]: Ah yes, I forgot to move this.  Will do it on next update

File: gather2-odk-importer/api/auth_utils.py [hunk 5]
```diff
@@ -0,0 +1,17 @@
+from django.http import HttpResponse
+from django.contrib.sites.models import Site
+
+
+def authorise(username=None, password=None):
```
Comment (readevalprint) [inline gather2-odk-importer/api/auth_utils.py]: :P British "s"

**Comment by readevalprint** [comment]
+1

File: gather2-odk-importer/api/auth_utils.py (hunk 5 above)
Comment (nkhumphreys) [inline gather2-odk-importer/api/auth_utils.py]: English?

**Pull request merged**